### PR TITLE
Close Popup/Dialog by clicking any mouse button outside

### LINKED ISSUE #4549: DropdownMenu doesn't close on right-click outside
Right-clicking outside an open `DropdownMenu` doesn't close it.

In Compose 1.5.x, right-clicking outside a dropdown menu would close it, but not get sent to the clicked element.

The ideal behavior would be to not only close the current dropdown menu, but also let the event through, so that a new dropdown menu can open at the new location.

**Affected platforms**
- Desktop

**Versions**
- Compose Multiplatform version*: 1.6.1

**To Reproduce**
```
fun main() = singleWindowApplication {
    val state = remember { DropdownMenuState() }
    Box(
        modifier = Modifier
            .fillMaxSize()
            .contextMenuOpenDetector(state)
    ) {
        DropdownMenu(state) {
            DropdownMenuItem(onClick = {}, contentPadding = PaddingValues(10.dp)) {
                Text("Menu Item 1")
            }
        }
    }
}
```

Right-click to open a context menu, then right click-outside it. Nothing happens.

**Expected behavior**
The menu should close, and a new menu at the newly clicked location should open.

### PR DESCRIPTION
**Pull request by MatkovIvan**
## Proposed Changes

- Remove condition for primary mouse button to send onClickOutside event

## Testing

Test: `PopupTest`/`DialogTest`

## Issues Fixed

Fixes https://github.com/JetBrains/compose-multiplatform/issues/4549

### CODE REVIEW

**Comment by m-sasha** [comment]
1. I think `onMouseEventOutside` will get called on mouse-wheel events too. We don't want to close the layer on those.
2. What *do* we want to do with mouse-wheel events here? I'm not sure...
3. Why does `onMouseEventOutside` need to re-check `inBounds`. It would be better if `DetectEventOutsideLayer` checked it before calling (given the function's name).

**Comment by m-sasha** [comment]
Also, let's add unit tests for all these scenarios.

**Comment by MatkovIvan** [comment]
1. It's already checked (not in changed code)
2. Left or right instead of any? 🤔 ok, probably makes sense
3. It's called not only from there, but yes it looks like we can do that

> let's add unit tests for all these scenarios.

As mentioned in the description - TBD. That's why it's in draft

**Comment by m-sasha** [comment]
> 1. It's already checked (not in changed code)

Ah, right, I see it only responds to mouse pressed/released events.

> Left or right instead of any? 🤔 ok, probably makes sense

Not what I meant. The layer should probably close on any button.

I'm just wondering what should be done on mouse-wheel events when they happen outside the layer bounds. Should we send them to the layers below? Do we need yet another layer flag here? Maybe something to consider & discuss later.

> It's called not only from there, but yes it looks like we can do that

Either that, or rename the function.

File: compose/ui/ui/src/skikoTest/kotlin/androidx/compose/ui/window/PopupTest.kt [hunk 1]
```diff
@@ -599,6 +599,40 @@ class PopupTest {
...
+        val buttons = PointerButtons(
+            isSecondaryPressed = true
+        )
+        scene.sendPointerEvent(
+            PointerEventType.Press,
+            position = Offset(10f, 10f),
+            buttons = buttons,
+            button = PointerButton.Secondary
+        )
+
```
Comment (m-sasha) [inline compose/ui/ui/src/skikoTest/kotlin/androidx/compose/ui/window/PopupTest.kt:632]: Should send a release here too

**Review by m-sasha — APPROVED** [review]

File: compose/ui/ui/src/skikoTest/kotlin/androidx/compose/ui/window/PopupTest.kt (hunk 1 above)
Comment (MatkovIvan) [reply compose/ui/ui/src/skikoTest/kotlin/androidx/compose/ui/window/PopupTest.kt:632]: No, it's on purpose - `Popup` should be closed on down event

Comment (m-sasha) [reply compose/ui/ui/src/skikoTest/kotlin/androidx/compose/ui/window/PopupTest.kt:632]: So dialogs close on release, but popups close on press? I see that in the code, but is there a reason for it?

Comment (MatkovIvan) [reply compose/ui/ui/src/skikoTest/kotlin/androidx/compose/ui/window/PopupTest.kt:632]: Yes. TLDR - Android

**Pull request merged**